internal/operators/cnv/cnv_operator.go

```diff
@@ -0,0 +1,135 @@
+
+import (
+	"context"
+	"encoding/json"
+	"strconv"
+
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/internal/hardware"
+	"github.com/openshift/assisted-service/internal/operators/api"
+	"github.com/openshift/assisted-service/internal/operators/lso"
+	models "github.com/openshift/assisted-service/models"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	masterMemory int64 = 150
+	masterCPU    int64 = 4
+	workerMemory int64 = 360
+	workerCPU    int64 = 2
+)
+
+type operator struct {
+	log logrus.FieldLogger
+}
+
+var Operator models.MonitoredOperator = models.MonitoredOperator{
+	Name:           "cnv",
+	OperatorType:   models.OperatorTypeOlm,
+	TimeoutSeconds: 60 * 60,
+}
+
+// NewCNVOperator creates new instance of a Container Native Virtualization installation plugin
+func NewCNVOperator(log logrus.FieldLogger) *operator {
+	return &operator{
+		log: log,
+	}
+}
+
+// GetName reports the name of an operator this Operator manages
+func (o *operator) GetName() string {
+	return Operator.Name
+}
+
+// GetDependencies provides a list of dependencies of the Operator
+func (o *operator) GetDependencies() []string {
+	return []string{lso.Operator.Name}
+}
+
+// GetClusterValidationID returns cluster validation ID for the Operator
+func (o *operator) GetClusterValidationID() string {
+	return string(models.ClusterValidationIDCnvRequirementsSatisfied)
+}
+
+// GetHostValidationID returns host validation ID for the Operator
+func (o *operator) GetHostValidationID() string {
+	return string(models.HostValidationIDCnvRequirementsSatisfied)
+}
+
+// ValidateCluster always return "valid" result
+func (o *operator) ValidateCluster(ctx context.Context, cluster *common.Cluster) (api.ValidationResult, error) {
+	// No need to validate cluster because it will be validate on per host basis
+	return api.ValidationResult{Status: api.Success, ValidationId: o.GetClusterValidationID()}, nil
+}
+
+// ValidateHost returns validationResult based on node type requirements such as memory and cpu
+func (o *operator) ValidateHost(ctx context.Context, cluster *common.Cluster, host *models.Host) (api.ValidationResult, error) {
+	var inventory models.Inventory
+	if err := json.Unmarshal([]byte(host.Inventory), &inventory); err != nil {
+		o.log.Errorf("Failed to get inventory from host with id %s", host.ID)
+		return api.ValidationResult{Status: api.Failure, ValidationId: o.GetClusterValidationID()}, err
+	}
+	// TODO: validate available devices on worker node like gpu and sr-iov and check whether there is enough memory to support them
+	if host.Role == models.HostRoleWorker {
+		cpu, _ := o.GetCPURequirementForWorker(ctx, cluster)
+		if inventory.CPU.Count < cpu {
+			return api.ValidationResult{Status: api.Failure, ValidationId: o.GetClusterValidationID(), Reasons: []string{"Insufficient CPU to deploy CNV. Required CPU count is: " + strconv.FormatInt(cpu, 10) + " but found: " + strconv.FormatInt(inventory.CPU.Count, 10)}}, nil
```

Comment:
I thought that you were going to validate CPU and memory before returning the `ValidateResult` object, so to consolidate both potential errors.

Reply:
By looking how it is [done](https://github.com/openshift/assisted-service/blob/master/internal/operators/ocs/validations.go#L48) for OCS there is no consolidation of error messages.